wallet/how-to/use-non-evm-networks/starknet/create-a-simple-starknet-dapp.md

```diff
@@ -0,0 +1,373 @@
+- [Yarn](https://yarnpkg.com/)
+
+## 1. Set up the project
+
+Create a new React project with TypeScript and set up the necessary dependencies for working with Starknet.
+
+### 1.1 Create a new React project with TypeScript
+
+Create a new directory called `get-starknet-tutorial` with a basic React app using TypeScript:
+
+```bash
+yarn create react-app get-starknet-tutorial --template typescript
+```
+    
+Change into the project directory:
+ 
+```bash
+cd get-starknet-tutorial
+```
+
+### 1.2 Add `get-starknet` and `starknet.js`
+
+Add the required libraries for interacting with Starknet:
+
+```bash
+yarn add get-starknet@3.3.0 starknet@6.11.0
+```
+
+<details>
+  <summary>Did you get an error?</summary>
+  <div>
+
+See how to [troubleshoot](troubleshoot.md) connection issues when configuring your dapp.
+
+Alternatively, you can review [Connect to Starknet](connect-to-starknet.md#1-set-up-the-project) to set up
+your React TypeScript project and install the `get-starknet` library.
+
+  </div>
+</details>
+
+Your file structure should look similar to the following:
+
+```text
+get-starknet-dapp/
+├── public/
+│   └── index.html
+└── src/
+    ├── components/
+    │   ├── WalletConnect.tsx
+    ├── App.tsx
+    ├── index.tsx
+    └── App.css
+```
+
+## 2. Configure the wallet connection
+
+Use the `connect` function to link your dapp with a user’s MetaMask wallet.
+
+### 2.1. Connect to MetaMask
+
+The `connect` function from `get-starknet` is the primary way to connect your dapp to a user's MetaMask wallet. 
+It opens a connection to MetaMask and returns an object containing important details about the wallet, such as the following:
+
+- `name`: The name of the wallet.
+- `icon`: The wallet's icon, which displays the wallet's logo.
+- `account`: The account object from `starknet.js`, which contains the wallet's address and provides access to account-specific operations.
+
+To import the necessary functions and connect to a wallet, add the following code to `src/App.tsx`:
+
+```typescript title="App.tsx"
+import { connect, type ConnectOptions } from "get-starknet";
+
+async function handleConnect(options?: ConnectOptions) {
+  const res = await connect(options);
+  // Access wallet details such as name, address, and icon
+  console.log(res?.name, res?.account?.address, res?.icon);
+}
+```
+
+### 2.2. Configure connection options
+
+`connect` accepts an optional `ConnectOptions` object. 
+This object can control the connection process, including:
+
+- `modalMode`: Determines how the connection modal behaves. The options are:
+  - `"alwaysAsk"`: Prompts the user every time a connection is initiated.
+  - `"neverAsk"`: Attempts to connect without showing the modal.
+- `modalTheme`: Sets the visual theme of the connection modal. The options are `"dark"` and `"light"`.
+
+You can configure these options as follows:
+
+```typescript
+handleConnect({ modalMode: "alwaysAsk", modalTheme: "dark" });
+```
+
+### 2.3. Create a `WalletAccount`
```

Comment:
@khanti42 this entire section is using Wallet Account. DO we have to rework all of this tutorial content?